### PR TITLE
NOISSUE - Add group tests to users and things service

### PR DESCRIPTION
**Pull request by felixgateru**
# What type of PR is this?
This is a feature as it adds groups tests to users and things service using the groups service mocks.

## What does this do?
This adds groups tests to users and things services.

## Which issue(s) does this PR fix/relate to?
None.

## Have you included tests for your changes?
No since this is the test file.

## Did you document any new/modified feature?
Yes, documentation in in code comments.

### CODE REVIEW

File: things/api/http/endpoints_test.go [hunk 1]
```diff
@@ -1947,12 +1940,826 @@ func TestListMembers(t *testing.T) {
...
+
+		svcCall := gsvc.On("Unassign", mock.Anything, tc.token, tc.groupID, mock.Anything, "users", mock.Anything).Return(tc.err)
+		res, err := req.make()
+		assert.Nil(t, err, fmt.Sprintf("%s: unexpected error %s", tc.desc, err))
+		assert.Equal(t, tc.status, res.StatusCode, fmt.Sprintf("%s: expected status code %d got %d", tc.desc, tc.status, res.StatusCode))
+		svcCall.Unset()
+	}
+}
+
+func TestAssignUserGroups(t *testing.T) {
```
Comment (rodneyosodo) [inline things/api/http/endpoints_test.go]:
```suggestion
func TestAssignGroupsToChannel(t *testing.T) {
```

File: things/api/http/endpoints_test.go [hunk 2]
```diff
@@ -1947,12 +1940,826 @@ func TestListMembers(t *testing.T) {
...
+
+		svcCall := gsvc.On("Assign", mock.Anything, tc.token, tc.groupID, mock.Anything, "channels", mock.Anything).Return(tc.err)
+		res, err := req.make()
+		assert.Nil(t, err, fmt.Sprintf("%s: unexpected error %s", tc.desc, err))
+		assert.Equal(t, tc.status, res.StatusCode, fmt.Sprintf("%s: expected status code %d got %d", tc.desc, tc.status, res.StatusCode))
+		svcCall.Unset()
+	}
+}
+
+func TestUnassignUserGroups(t *testing.T) {
```
Comment (rodneyosodo) [inline things/api/http/endpoints_test.go]: same

File: things/api/http/endpoints_test.go [hunk 3]
```diff
@@ -1947,12 +1940,826 @@ func TestListMembers(t *testing.T) {
...
+
+		svcCall := gsvc.On("Unassign", mock.Anything, tc.token, tc.groupID, mock.Anything, "channels", mock.Anything).Return(tc.err)
+		res, err := req.make()
+		assert.Nil(t, err, fmt.Sprintf("%s: unexpected error %s", tc.desc, err))
+		assert.Equal(t, tc.status, res.StatusCode, fmt.Sprintf("%s: expected status code %d got %d", tc.desc, tc.status, res.StatusCode))
+		svcCall.Unset()
+	}
+}
+
+func TestConnectChannelThing(t *testing.T) {
```
Comment (rodneyosodo) [inline things/api/http/endpoints_test.go]:
```suggestion
func TestConnectChannelToThing(t *testing.T) {
```

File: things/api/http/endpoints_test.go [hunk 4]
```diff
@@ -1947,12 +1940,826 @@ func TestListMembers(t *testing.T) {
...
+
+		svcCall := gsvc.On("Assign", mock.Anything, tc.token, tc.channelID, "group", "things", []string{tc.thingID}).Return(tc.err)
+		res, err := req.make()
+		assert.Nil(t, err, fmt.Sprintf("%s: unexpected error %s", tc.desc, err))
+		assert.Equal(t, tc.status, res.StatusCode, fmt.Sprintf("%s: expected status code %d got %d", tc.desc, tc.status, res.StatusCode))
+		svcCall.Unset()
+	}
+}
+
+func TestDisconnectChannelThing(t *testing.T) {
```
Comment (rodneyosodo) [inline things/api/http/endpoints_test.go]: same

File: things/api/http/endpoints_test.go [hunk 5]
```diff
@@ -1947,12 +1940,826 @@ func TestListMembers(t *testing.T) {
...
+		},
+		{
+			desc:  "connect thing to channel with empty channel id",
+			token: validToken,
+			reqBody: groupReqBody{
+				ChannelID: "",
+				ThingID:   validID,
+			},
+			contentType: contentType,
+			status:      http.StatusUnprocessableEntity,
```
Comment (rodneyosodo) [inline things/api/http/endpoints_test.go]: This should be code 400 change validation to return missing id. Same as other cases

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

**Review by rodneyosodo — DISMISSED** [review]
LGTM

File: things/api/http/endpoints_test.go [hunk 6]
```diff
@@ -1947,12 +1940,826 @@ func TestListMembers(t *testing.T) {
...
+		desc        string
+		token       string
+		groupID     string
+		reqBody     interface{}
+		contentType string
+		status      int
+		err         error
+	}{
+		{
+			desc:    "unassign users to group successfully",
```
Comment (dborovcanin) [inline things/api/http/endpoints_test.go]: unassign users from a group successfully

Comment (dborovcanin) [reply things/api/http/endpoints_test.go]: Fux other cases, as well.

File: users/api/endpoint_test.go [hunk 7]
```diff
@@ -2890,16 +2884,413 @@ func TestListUsersByThingID(t *testing.T) {
...
+	cases := []struct {
+		desc    string
+		token   string
+		groupID string
+		reqBody interface{}
+		status  int
+		err     error
+	}{
+		{
+			desc:    "assign groups to group successfully",
```
Comment (dborovcanin) [inline users/api/endpoint_test.go]: assign group to a parent group successfully

Comment (dborovcanin) [reply users/api/endpoint_test.go]: Use `assign group to a parent group` instead of `assign group to a group`.

**Review by dborovcanin — CHANGES_REQUESTED** [review]

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**